subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -542,6 +542,11 @@ static ssize_t write_key_based_pairing(struct bt_conn *conn,
 	if (net_buf_simple_max_len(&gatt_write) == FP_CRYPTO_ECDH_PUBLIC_KEY_LEN) {
 		keygen_params.public_key = net_buf_simple_pull_mem(&gatt_write,
 								   FP_CRYPTO_ECDH_PUBLIC_KEY_LEN);
+	} else if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_SUBSEQUENT_PAIRING)) {
```

Comment:
I wonder if we shouldn't move this logic to `fp_keys_generate_key` (it could return `-ENOTSUP` and we could properly handle this error code here). The `fp_keys` is the module responsible for managing Fast Pair keys and pairing procedures. The `fp_gatt_service` is responsible for communication over GATT.

Reply:
the GATT request fields does not match our feature selection. I implemented this logic in the fp_keys.c at first but decided it would be better to exit as soon as possible (after parsing the public key field)

Reply:
we will discuss it offline

Reply:
Stayed with the current logic for rejecting the subsequent pairing at the GATT Write request callback level.

Added an assert to the fp_keys module in case someone incorrectly uses the `fp_keys_generate_key` function in the future (currently this function is only used in the context of the GATT Write request callback for the Key-based Pairing characteristic).